### PR TITLE
Product import export command

### PR DESCRIPTION
**Pull request by eguzki**
Two new commands
* `3scale product export`: export 3scale product and used backends in [Product CRD](https://github.com/3scale/3scale-operator/blob/3scale-2.10.0-CR2/doc/product-reference.md) format and [Backend CRD](https://github.com/3scale/3scale-operator/blob/3scale-2.10.0-CR2/doc/backend-reference.md) format
* `3scale product import`: import 3scale product and used backends in [Product CRD](https://github.com/3scale/3scale-operator/blob/3scale-2.10.0-CR2/doc/product-reference.md) format and [Backend CRD](https://github.com/3scale/3scale-operator/blob/3scale-2.10.0-CR2/doc/backend-reference.md) format

Requires enhancements from #285

### CODE REVIEW

File: lib/3scale_toolbox/commands/product_command/import_command.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        def run
+          validate_artifacts_resource!
+
+          product_list.each do |product|
+            context = {
+                target_remote: remote,
+                source_remote: crd_remote,
+                source_service_ref: product.system_name,
+                logger: Logger.new('/dev/null')
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/commands/product_command/import_command.rb:31]: Is this intended to be /dev/null?

File: lib/3scale_toolbox/crds/application_plan_dump.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module ThreeScaleToolbox
+  module CRD
+    module ApplicationPlan
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/crds/application_plan_dump.rb:3]: Should this be called  `ApplicationPlanSerializer` to follow naming convention?

File: lib/3scale_toolbox/crds/backend_parser.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+module ThreeScaleToolbox
+  module CRD
+    class BackendParser
+      Metric = Struct.new(:system_name, :friendly_name, :description, :unit)
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/crds/backend_parser.rb:4]: Here we are defining our own custom structs.

There are their corresponding 'Entity' classes, can't we reuse them? some classes on the model will be using this custom structs and others will be using the entities one, which is kind of confusing.

File: lib/3scale_toolbox/commands/product_command/import_command.rb (hunk 1 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/product_command/import_command.rb:31]: Yes, for backwards compatibility, the `product copy` command still prints messages. But for the new command, we do not want those messages to be printed. Instead, the new command generates in stdout a report in json format with structured info.

File: lib/3scale_toolbox/crds/backend_parser.rb (hunk 3 above)
Comment (eguzki) [reply lib/3scale_toolbox/crds/backend_parser.rb:4]: This `Metric` struct is private for the `BackendParser` and should not be used externally. 

The entities are interface to external remote objects and have methods and required input parameters (like remote) that do not make sense for this structs. They only share the list of attributes.

File: lib/3scale_toolbox/crds/application_plan_dump.rb (hunk 2 above)
Comment (eguzki) [reply lib/3scale_toolbox/crds/application_plan_dump.rb]: done

File: lib/3scale_toolbox/commands/product_command/import_command.rb (hunk 1 above)
Comment (miguelsorianod) [reply lib/3scale_toolbox/commands/product_command/import_command.rb:31]: :+1:

File: lib/3scale_toolbox/crds/backend_parser.rb (hunk 3 above)
Comment (miguelsorianod) [reply lib/3scale_toolbox/crds/backend_parser.rb:4]: :+1:

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**